edenai_apis/apis/sightengine/sightengine_api.py

```diff
@@ -0,0 +1,171 @@
+            api_keys=api_keys or {},
+        )
+        self.api_url = "https://api.sightengine.com/1.0"
+        self.headers = {
+            "Content-Type": "application/json",
+            "Authorization": f'Bearer {self.api_settings["api_key"]}',
+            
+        }
+        self.webhook_settings = load_provider(ProviderDataEnum.KEY, "webhooksite")
+        self.webhook_token = self.webhook_settings["webhook_token"]
+        self.webhook_url = f"https://webhook.site/{self.webhook_token}"
+
+    def image__deepfake_detection(
+        self, file: Optional[str] = None, file_url: Optional[str] = None
+    ) -> ResponseType[ImageDeepfakeDetectionDataclass]:
+        if not file_url and not file:
+            raise ProviderException("file or file_url required")
+
+        payload = {"url": file_url or upload_file_to_s3(file, file),
+                              "models":"deepfake",
+                              "api_user": self.api_settings["api_user"],
+                              "api_secret": self.api_settings["api_key"]}
+        
+        response = requests.get(
+            f"{self.api_url}/check.json",
+            params=payload,
+        )
+
+        print(response.json())
+
+        if response.status_code != 200:
+            raise ProviderException(response.json(), code=response.status_code)
+
+        original_response = response.json()
+
+        score = 1 - extract(original_response,["type","deepfake"],None)
+        if score is None:
+            raise ProviderException(response.json())
+        prediction = ImageDeepfakeDetectionDataclass.set_label_based_on_score(score)
+
+        standardized_response = ImageDeepfakeDetectionDataclass(
+            ai_score=score,
+            prediction=prediction,
+        )
+
+
+        return ResponseType[ImageDeepfakeDetectionDataclass](
+            original_response=original_response,
+            standardized_response=standardized_response,
+        )
+    
+    def video__deepfake_detection_async__launch_job(
+        self, file: Optional[str] = None, file_url: Optional[str] = None
+    ) -> AsyncLaunchJobResponseType:
+        if not file_url and not file:
+            raise ProviderException("file or file_url required")
+        
+        payload = {
+            "models": "deepfake",
+            "api_user": self.api_settings["api_user"],
+            "api_secret": self.api_settings["api_key"],
+            "callback_url": self.webhook_url,
+        }
+
+        if file:
+            with open(file, 'rb') as video_file:
+                files = {'media': video_file}
+
+                response = requests.post(
+                    f"{self.api_url}/video/check.json",
+                    files=files,
+                    data=payload,
+                )
+        elif file_url:
+            payload['stream_url'] = file_url
+
+            response = requests.get(
+                f"{self.api_url}/video/check.json",
+                params=payload,
+            )

```

Comment:
You can here maybe construct the requests parameters dynamically and than do only one request call depending on the method (get or post) using:
```python
requests.request(method, url, **kwargs)
```

Here also adding a try/catch block